src/lcm/grids.py

```diff
@@ -46,7 +46,33 @@ def __post_init__(self) -> None:
 
     def to_jax(self) -> Array:
         """Convert the grid to a Jax array."""
-        return jnp.array(list(self.options))
+        return jnp.array(_get_fields(self.options))
```

Comment:
This seems too late to throw an error, why not do that during validation / use the same code?